federation/cluster/common.sh

```diff
@@ -24,7 +24,14 @@
 
 : "${KUBE_ROOT?Must set KUBE_ROOT env var}"
 
+# Provides the $KUBERNETES_PROVIDER variable and detect-project function
+source "${KUBE_ROOT}/cluster/kube-util.sh"
+
+# Populates $PROJECT
+detect-project
+
 FEDERATION_IMAGE_REPO_BASE=${FEDERATION_IMAGE_REPO_BASE:-'gcr.io/google_containers'}
+FEDERATION_PUSH_REPO_BASE=${FEDERATION_PUSH_REPO_BASE:-gcr.io/${PROJECT}}
```

Comment:
Yeah, this doesn't support domain-scoped project names. I think it is fair to expect people to set `$FEDERATION_PUSH_REPO_BASE` if the project name isn't a plain string. This is a convenience for people who do have plain string project names.